src/util.cc

```diff
@@ -368,9 +368,9 @@ std::string ToLower(std::string in) {
   return in;
 }
 
-bool CompareString(const std::string& lhs, const std::string& rhs, bool case_insensitive) {
-    return lhs.size() == rhs.size() && std::equal(lhs.begin(), lhs.end(), rhs.begin(),[case_insensitive](const char& l, const char& r) {
-        if (l == r || case_insensitive && std::tolower(l) == std::tolower(r)) {
+bool CaseInsensitiveCompare(const std::string& lhs, const std::string& rhs) {
+    return lhs.size() == rhs.size() && std::equal(lhs.begin(), lhs.end(), rhs.begin(),[](const char& l, const char& r) {
+        if (l == r || std::tolower(l) == std::tolower(r)) {
             return true;
         }
         return false;
```

Comment:
```suggestion
        return std::tolower(l) == std::tolower(r);
```

Reply:
but is it possible to reduce function calls here?

Reply:
I think the probability of `l != r` is higher than `l == r` for normal string, so the current code may walk more branches, which is bad for pipelining and branch prediction. And, unless an obvious benchmark can be provided, I prefer simpler code.